File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.py
arr = ['c=', 'c-', 'dz=', 'd-', 'lj', 'nj', 's=', 'z=']
word = input()
word += '0'

ans = 0
while True:
    if len(word) == 1:
        break

    ans += 1
    if word[:2] in arr:
        word = word[2:]
    elif word[:3] in arr:
        word = word[3:]
    else:
        word = word[1:]

print(ans)

File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) {
		String[] arr = {"c=", "c-", "dz=", "d-", "lj", "nj", "s=", "z="};
		List<String> strList = new ArrayList<>(Arrays.asList(arr));

		Scanner sc = new Scanner(System.in);
		
		String str = sc.next();
		
		int s = 0;
		int ans = 0;
		
		while (s < str.length()) {
			
			if (s + 2 <= str.length() && strList.contains(str.substring(s, s+2))) {
				ans += 1;
				s += 2;
			} else if (s + 3 <= str.length() && strList.contains(str.substring(s, s+3))) {
				ans += 1;
				s += 3;
			} else {
				ans += 1;
				s += 1;
			}
		}
		
		System.out.println(ans);
	}

}
